mozetl/search/aggregates.py

```diff
@@ -222,15 +222,20 @@ def _drop_source_columns(base):
         return derived
 
     def _select_counts(main_summary, col_name, count_udf=None):
-        derived = main_summary.withColumn(
-            "single_search_count", explode(col("search_counts"))
-        ).filter("single_search_count.count < %s" % MAX_CLIENT_SEARCH_COUNT)
+        ad_fields = ("ad_click", "search_with_ads")
+        if col_name in ad_fields:
+            # special case for ad click probes
+            derived = main_summary
+        else:
+            derived = main_summary.withColumn(
+                "single_search_count", explode(col("search_counts"))
+            ).filter("single_search_count.count < %s" % MAX_CLIENT_SEARCH_COUNT)
         if count_udf is not None:
             derived = derived.withColumn(col_name, explode(count_udf))
         derived = _drop_source_columns(
             derived.select(["*"] + _get_search_fields(col_name)).drop(col_name)
         )
-        if col_name is not "single_search_count":
+        if col_name is not "single_search_count" or col_name in ad_fields:
```

Comment:
I don't think this extra logic is necessary. If the col_name is one of the ad fields, by definition it is not "single_search_count".

Reply:
True, this will just drop a phantom column in the case of the ad_clicks, but I was being (perhaps too) explicit.

Reply:
Actually, I think this if-statement can be omitted altogether with the added logic at the top of the function.